kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/kroxylicious/KroxyliciousConfigMapTemplates.java

```diff
@@ -46,16 +53,47 @@ public static ConfigMapBuilder defaultKroxyliciousConfig(String clusterName, Str
      *
      * @param clusterName the cluster name
      * @param namespaceName the namespace name
-     * @param topicName the topic name
-     * @param config config
+     * @param testKmsFacade the test kms facade
      * @return the config map builder
      */
-    public static ConfigMapBuilder kroxyliciousRecordEncryptionConfig(String clusterName, String namespaceName, String topicName, Config config) {
+    public static ConfigMapBuilder kroxyliciousRecordEncryptionConfig(String clusterName, String namespaceName, TestKmsFacade<?, ?, ?> testKmsFacade) {
         return baseKroxyliciousConfig(namespaceName)
-                .addToData("config.yaml", getRecordEncryptionConfigMap(clusterName, topicName, config));
+                .addToData("config.yaml", getRecordEncryptionConfigMap(clusterName, testKmsFacade));
+    }
+
+    private static String buildEncryptionFilter(TestKmsFacade<?, ?, ?> testKmsFacade) {
+        return """
+                - type: RecordEncryption
+                  config:
+                    kms: %s
+                    kmsConfig:
+                      %s
+                    selector: TemplateKekSelector
+                    selectorConfig:
+                      template: "${topicName}"
+                """.formatted(testKmsFacade.getKmsServiceClass().getSimpleName(), getNestedYaml(testKmsFacade.getKmsServiceConfig(), 6));
     }
 
-    private static String getRecordEncryptionConfigMap(String clusterName, String topicName, Config config) {
+    private static String getNestedYaml(Object config, int indent) {
+        String configYaml;
+        YAMLFactory factory = YAMLFactory.builder()
```

Comment:
I think this could have stayed at field level (possibly via a static initialiser) or be initialised once in the constructor. Creating objectmappers is relatively expensive so best avoided.

Reply:
Good to know. Refactored, thanks!